ando/newchecker.py

```diff
@@ -0,0 +1,160 @@
+import os
+import os.path as op
+import re
+import argparse
+
+# construct the set of rules to be applied, level by level
+rules_set = []
+# level 0
+currentdepth_rules={}
+currentdepth_rules['authorized_folders'] = ['exp-([a-zA-Z0-9]+)']
+currentdepth_rules['authorized_files'] = ['subject\\.tsv','dataset_description\\.tsv', 'bouh.tsv']
+currentdepth_rules['mandatory_files'] = ['subject\\.tsv','dataset_description\\.tsv']
+rules_set.append(currentdepth_rules)
+# level 1
+currentdepth_rules={}
+currentdepth_rules['authorized_folders'] = ['sub-([a-zA-Z0-9]+)']
+currentdepth_rules['authorized_files'] = []
+currentdepth_rules['mandatory_files'] = []
+rules_set.append(currentdepth_rules)
+# level 2
+currentdepth_rules={}
+currentdepth_rules['authorized_folders'] = ['ses-([a-zA-Z0-9]+)']
+currentdepth_rules['authorized_files'] = []
+currentdepth_rules['mandatory_files'] = []
+rules_set.append(currentdepth_rules)
+# level 3
+currentdepth_rules={}
+currentdepth_rules['authorized_folders'] = ['ephys']
+currentdepth_rules['authorized_files'] = ['sub-([a-zA-Z0-9]+)_ses-([a-zA-Z0-9]+)([\w\\-]*)_ephys\\.nix',
+                               'sub-([a-zA-Z0-9]+)_ses-([a-zA-Z0-9]+)([\w\\-]*)_ephys\\.nxi',
+                               'sub-([a-zA-Z0-9]+)_ses-([a-zA-Z0-9]+)([\w\\-]*)_ephys\\.nwb']
+currentdepth_rules['mandatory_files'] = []
+rules_set.append(currentdepth_rules)
+
+
+
+def newchecker(input_directory):
+    """ Checks the ephys-BIDS validity of a data set.
+
+    The specifications that define what is checked by this function is available in the following document:
+
+
+    Parameters
+    ----------
+        directory : string
+            Name of the root directory containing the data set to be checked
+
+    Returns
+    -------
+        valid : boolean
+            True if the data set follows the ephys-BIDS specification; False if not
+
+        error_list : list
+            List of errors (empty if the data set is valid)
+    """
+
+
+    error_list = []
+
+    # walk through the directory tree using the os.walk function
+    for ind, (root, dirs, files) in enumerate(os.walk(input_directory)):
+        # compute depth of the directory given as input
+        if ind == 0:
+            # remove ending / or \ if the input directory was given with it at the end
+            if root.endswith(os.path.sep):
+                root = root[:-1]
+            # count the number of / or \ in the directory name to estimate its "depth"
+            initial_depth = root.count(os.path.sep)
+        # estimate at which level we are
+        depth = root.count(os.path.sep) - initial_depth
```

Comment:
I think this code could be simplified and more robust by using the pathlib library. Then these lines could look like this 
```suggestion
            # remove ending / or \ if the input directory was given with it at the end
            root = pathlib.Path(root)
            # count the number of / or \ in the directory name to estimate its "depth"
            initial_depth = len(root.resolve().parents)
            
        # estimate at which level we are
        depth = len(pathlib.Path(root).resolve().parents) - initial_depth
```